dev_tools/oneapi/convert_source.sh

```diff
@@ -0,0 +1,429 @@
+                    num=$((num-1))
+                else
+                    if [ -n "${var}" ]; then
+                        var="${var},"
+                    fi
+                    var="${var}${temp}"
+                fi
+                temp=""
+            fi
+        done
+        # echo "// var ${var}"
+        # echo "var ${var}"
+        if [[ "${num}" -lt 2 ]]; then
+            var="Error"
+        else
+            if [[ ${num} -eq 2 ]]; then
+                var="${var}, 0"
+                num=$((num+1))
+            fi
+            if [[ ${num} -eq 3 ]]; then
+                # var="${var}, exec->get_queue()"
+                var="${var}, GET_QUEUE"
+                num=$((num+1))
+            fi
+        fi
+        local suffix=""
+        local function_name=$(echo "${BASH_REMATCH[1]}" | sed -E 's/(.*::)| //g')
+        # echo "// functioname ${function_name}"
+        MAP_FILE="map_list"
+        suffix=$(cat "${MAP_FILE}" | sed -nE "s/${function_name} -> ${function_name}(.*)/\1/p")
+        suffix_matches=$(echo "${suffix}" | wc -l)
+        if [[ "${suffix_matches}" -gt 1 ]]; then
+            echo "static_assert(false, \"Has ${suffix_matches} matches for ${function_name}\");"
+            suffix=$(echo "${suffix}" | head -1)
+        fi
+
+        if [[ "${suffix}" = *"${CONFIG_SELECTION_SUFFIX}"* ]]; then
+            var="${var}, exec"
+        fi
+        local result="${BASH_REMATCH[1]}${suffix}${BASH_REMATCH[2]}(${var})${BASH_REMATCH[4]}"
+        result=$(echo "${result}" | sed -E 's/\) *\(+/,/g;s/, *\)/\)/g')
+        if [[ "${suffix}" = "" ]]; then
+            # if the function does not exist in the file, comment it.
+            echo "//remove//${result}"
+        else
+            echo "${result}"
+        fi
+    fi
+}
+
+
+convert_regex_allowed() {
+    local str="$1"
+    str=$(echo "${str}" | sed -E 's/(\/|\(|\)|\.)/\\\1/g')
+    echo "$str"
+}
+
+# Transfer header file to the correct one
+# common -> dpcpp_common
+# cuda -> dpcpp
+# add sycl header
+
+input="$1"
+filename="${ROOT_DIR}/$input"
+if [[ "${VERBOSE}" == 1 ]]; then
+    echo "Porting file ${filename}"
+fi
+# check file exists
+
+if [ ! -f "${filename}" ]; then
+    echo "${filename} does not exist"
+    exit 1
+fi
+
+
+temp=""
+IN_SYNTAX="false"
+DEVICE_FILE=""
+GINKGO_LICENSE_BEACON="******************************<GINKGO LICENSE>******************************"
+UNFORMAT_FILE="unformat.cpp"
+FORMAT_FILE="format.cpp"
+cp "${filename}" "${UNFORMAT_FILE}"
+OUTPUT_FILE="source.cpp"
+EMBED_FILE="embed.cu"
+EMBED_HOST_FILE="embed_host.cu"
+if [[ "${VERBOSE}" == 1 ]]; then
+    echo "the original file ${UNFORMAT_FILE}"
+    echo "the formatted file ${FORMAT_FILE}"
+    echo "collect common/*.inc in file ${EMBED_FILE}"
+    echo "add autohost func in file ${EMBED_HOST_FILE}"
+    echo "convert original CUDA call in file ${OUTPUT_FILE}, which is the file for dpct"
+fi
+rm "${OUTPUT_FILE}"
+# echo "#include \"trick/dim3.hpp\"" >> "${OUTPUT_FILE}"
```

Comment:
It's the same in other places, if lines are not needed anymore it's better to remove them.